### PR TITLE
React Capstone project

### PR DESCRIPTION
**Pull request by sawmon71293**
## In this project, I did the following:

- Add fish store 
- Style using MUI
- Connect with API
- Connect React and Fish store
- Use  useParams to carry ID
- Create fishProfiles page filter by the region
- Create Detail page with fish facts
- Use React Router for SPA

### CODE REVIEW

File: src/setupTests.js [hunk 1]
```diff
@@ -1,5 +0,0 @@
-// jest-dom adds custom jest matchers for asserting on DOM nodes.
```
Comment (Alfredbis29) [inline src/setupTests.js:1]: - [ ] As required for this project **App has unit tests for several components of the app and some pure functions in Redux code**., but  I noticed that there are currently no `unit tests` in your app. As you may know, unit tests are an important part of the development process, as they help to ensure the quality and reliability of the code. Kindly add them :wink:  for your project to match the requirements :wink: 

Unit tests can help to catch errors and bugs early on in the development process, which can save a lot of time and effort in the long run. They also help to ensure that the app meets certain requirements and specifications and that changes made to the code do not have unintended consequences.

_Note: By implementing unit tests in the app, we can improve the overall quality of the code and ensure that it functions correctly and as expected. I would be happy to help with writing and implementing these tests please :+1:_

File: README.md [hunk 2]
```diff
@@ -112,7 +99,12 @@ You can deploy this project using:
 
 ### Live <a name="live"></a>
 
-- This page is yet to be deployed
+- Live Demo: 
+[fishwatch](https://fishwatch.onrender.com/)
+
+### Video <a name="video"></a>
+- Live Video:
```
Comment (Alfredbis29) [inline README.md:106]: - [ ] I can see that you record a good video but short and your record is missing the following ones as required for this project :point_down: 

- The maximum length of the video is 5 minutes. but for you is 3min
-  The video contains a clear and objective description of the project in professional English. The topic and purpose of the project are made clear. `kindly improve this `
-  The video contains a demo of the project, highlighting all the project features. You didn't perform that :wink: :smile: 
-  The video contains a highlight about a piece of the code the student finds interesting or is proud of making. you didn't talk about the codes you made during this project and that you feel proud of  :upside_down_face: 
Review your [Video](https://www.loom.com/share/69f5276de20b4ac1928f7b1c074336e1) and improve :wink:

File: src/App.js [hunk 3]
```diff
@@ -1,8 +1,20 @@
 import './App.css';
+import { Routes, Route } from 'react-router-dom';
+
+import FishProfiles from './components/FishProfiles/FishProfiles';
+import NavBar from './components/Nav/NavBar';
+import Detail from './components/Detail/Detail';
```
Comment (Alfredbis29) [inline src/App.js:7]: - [ ] Good job so far but your app is running with many errors, I noticed that there were several `modules` not found errors in your React application. The errors appear to be related to missing dependencies such as `react-router-dom`, , `react-redux`, and `react-icons/fa` I can see them in your dependencies but I don't know what can be the issue , seemed you didn't setup well your react app well. These dependencies are necessary for certain components in your application, such as `Detail.js`,`FishProfiles.js`, and `NavBar.js`. kindly fix this error :point_down:  then ask for the second opinion when you are ready :wink: 
Screenshot from 2023-05-06 13-06-37

**Review by Alfredbis29 — CHANGES_REQUESTED** [review]
#  STATUS: REQUIRED CHANGES 
 ## Hi @sawmon71293 
 Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

  
 ## Highlights 
:heavy_check_mark: You have a good code structure.
:heavy_check_mark: Your project looks professional
 
 ## Required Changes 

- [ ]  _Check Under the review_


 ## Optional suggestions 

- [ ]  Kindly  follow the requirement of this project and will get there , but all is fine 

 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._
 
 Cheers and Happy coding! :clap: :clap: :clap:
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Alfredbis29 in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @sawmon71293 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is professional :heavy_check_mark: 

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by sawmon71293** [comment]
Thanks for your thorough reviews. It was a great learning opportunity. @tresorsawasawa  and @Alfredbis29 .

**Pull request merged**